### DISCUSSION TITLE
option to select 8 bit or 16 bit

### DISCUSSION THREAD

**Discussion by kaelsonofkrypto** [nagadomi/nunif#171] [General]
Hi is it possible to add an option where we can choose 16 bit or 8bit png

**Comment by nagadomi** [comment]
What file are you talking about exactly?

**Reply by kaelsonofkrypto** [reply]
So  for the export disparity function at first i thought the pngs were converting at 8 bit and the original request was 16 bit, but someone said that iw3 is converting at 16 bit. But then it dawned on me that it still would be nice to have The option for users to pick either or because I have seen a couple people who mention the file size was heavy for them. Preferably I prefer 16-bit because I want the highest quality as possible so I don't get any banding on the depth map  but I do understand some people who are trying to save space

**Reply by nagadomi** [reply]
8-bit depth is useless because it causes terrible banding.
I will not add that option.
If disk space is not enough, convert directly without using Export.

**Reply by kaelsonofkrypto** [reply]
> 8-bit depth is useless because it causes terrible banding. I will not add that option. If disk space is not enough, convert directly without using Export.

so today i did a test and jpeg wasnt bad as you think as long as you adjust certain settings and u do jpeg from davinci. If you choose the best option on jpeg in davinci, it yields a high quality jpeg while still being half the size of png. However here is what i did notice, with png you can safely do 1.6 depth with minimal artifacts, however with jpeg the upper limit is about 1.20 to 1.3 on depth more than likely because the lower compression of png allows u to get away with more. Jpeg is totally not useless but indeed if u go higher than 1.3 depth on jpeg, it will cause edge artifacts. Davinci allows u to do jpeg at yuv  444 8 bit, and the results was actually really comparable to png surprisingly. however without a great  jpg encoder or the jpeg 2000 format which is less lossless than regular jpeg, i could see why u would prefer to keep it png.  Jpeg however could still be a viable option if u choose the right settings and i have actually managed to get a pretty great quality out of it using davinci. But because i like the best quality, i will be sticking to PNG.

i have tested at iw3 at all resolutions from 480p to even 8k full 3d sbs which is equivalent to 16k, and i have even tested downscaling 8k to 720p, i have tested depth resolutions all the way from default to 2160. Some of what i have learned that having the depth resolution equal to the 2d or the depth resolution higher than the 2d source produces better depth.  however u cant do this natively in iw3 because the depth model of depth anything was only trained on 518, but u can do an upscale of the 518 map and it indeed will yield better results than doing just the 518 depth native.

what u have created is really special nevertheless.
